File: app/src/main/java/com/yhjoo/dochef/adapter/MainFragmentAdapter.java
package com.yhjoo.dochef.adapter;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.List;

public class MainFragmentAdapter extends FragmentPagerAdapter {
    /*
        TODO
        이건 곧 아래뷰 페이저로 바뀜
     */
    private final List<Fragment> fragmentList;

    public MainFragmentAdapter(FragmentManager fm, int behavior, List<Fragment> fragmentList) {
        super(fm, behavior);
        this.fragmentList = fragmentList;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }
}

File: app/src/main/java/com/yhjoo/dochef/ui/adapter/MainFragmentAdapter.kt
package com.yhjoo.dochef.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MainFragmentAdapter(
    fm: FragmentManager?,
    behavior: Int,
    private val fragmentList: List<Fragment>
) : FragmentPagerAdapter(
    fm!!, behavior
) {
    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }
}